src/libs/auth.js

```diff
@@ -0,0 +1,98 @@
+import _ from 'lodash/fp'
+import { Sam, Rawls, Leo } from 'src/libs/ajax'
+import * as Config from 'src/libs/config'
+import * as Utils from 'src/libs/utils'
+
+export const authStore = Utils.atom({
+  isSignedIn: undefined,
+  isRegistered: undefined
+})
+
+export const initializeAuth = _.memoize(async () => {
+  await new Promise(resolve => window.gapi.load('auth2', resolve))
+  await window.gapi.auth2.init({ clientId: await Config.getGoogleClientId() })
+  authStore.update(state => ({ ...state, isSignedIn: getAuthInstance().isSignedIn.get() }))
+  getAuthInstance().isSignedIn.listen(isSignedIn => {
+    return authStore.update(state => {
+      return { ...state, isSignedIn, isRegistered: isSignedIn ? state.isRegistered : undefined }
```

Comment:
`isRegistered: isSignedIn || state.isRegistered`?

Reply:
In this case we're using these as 'tri-state' booleans where `undefined` means we don't know yet. So the logic here is intentional: whenever we become signed out, our registration status becomes unknown (until the next time we check)